js/embed_media_back_button.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * @file
+ * Adds the back button for embeded media select modal in CKEditor.
+ * Everytime the user goes to different page within the media embed select modal
+ * it will save the filter values to retrive the search result when user
+ * comes back to the first page.
+ */
+
+ (function ($, Drupal) {
+  'use strict';
+
+  Drupal.behaviors.tide_media_browser_iframe = {
+    attach: function (context, settings) {
+      if (!window.CKEDITOR) {
+        return;
+      }
+      let iframePath, itemName, licenseType, mediaType, published, site
+      const iframeName = 'entity_browser_iframe_tide_media_browser_iframe'
+      const homePagePath = '/entity-browser/iframe/tide_media_browser_iframe'
+      const backButton = $('#entity-embed-dialog-tide-media-browser-home')
+      $(document).ready(function () {
+        // Remove the name field value on modal first load if any.
+        localStorage.removeItem("tideMediaBrowsernameFilterVal");
+        $("iframe").on("load", function() {
+          if ($("iframe") && $("iframe").length > 0) {
+            for (let i = 0; i < $("iframe").length; i++) {
+              if ($("iframe")[i].name && $("iframe")[i].name === iframeName) {
+                iframePath = $("iframe")[1].contentWindow.location.pathname
+                // Shows the back button when iframe is not in the first page.
+                if (iframePath && iframePath !== homePagePath) {
+                  backButton.css("display", "block");
+                }
+                // Hides the back button for the first page.
+                if (iframePath && iframePath === homePagePath) {
+                  backButton.css("display", "none");
+                }
+                // Get the filter values on iframe load.
+                itemName =  $(this).contents().find('#edit-name').val()
+                licenseType = $(this).contents().find('#edit-field-license-type-target-id-1 :selected').text()
+                mediaType = $(this).contents().find('#edit-bundle :selected').text()
+                published = $(this).contents().find('#edit-status :selected').text()
+                site = $(this).contents().find('#edit-field-media-site-target-id :selected').text()
+                // Trigger ajax call submit to retrive the user search for current session.
+                // Will only trigger if there is any filter value set for this session.
+                if (checkFilterValuesOnLoad (itemName, licenseType, mediaType, published, site)) {
+                  $(this).contents().find('#edit-submit-tide-media-browser').trigger('click');
```

Comment:
the filter search for media items is a ajax callback, for this reason, when going back to the first page if there are any filter values it will retrieve that but it won't retrieve the search results, this will make sure that if there are any pre-selected values in the filter, it will just trigger ajax callback to get the search results

Reply:
on empty values it won't get triggered